pumpkin/src/server/translation.rs

```diff
@@ -0,0 +1,208 @@
+#![expect(dead_code)]
+
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+use pumpkin_config::TranslationConfig;
+use pumpkin_core::text::{style::Style, TextComponent, TextContent};
+use serde_json::Value;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TranslationError {
+    #[error("File cannot be opened.")]
+    InvalidFile,
+    #[error("Failed to read file. Error: {0}")]
+    FileRead(std::io::Error),
+    #[error("Invalid JSON encountered. Use only objects in the translation file.")]
+    JsonParse,
+}
+
+pub enum Translations<'a> {
+    Untranslated(TextComponent<'a>),
+    Translated(HashMap<String, TextComponent<'a>>),
+}
+
+pub fn translate<'a>(
+    config: &TranslationConfig,
+    message: &'a str,
+) -> Result<Translations<'a>, TranslationError> {
+    if !config.enabled {
+        return Ok(Translations::Untranslated(TextComponent::text(message)));
+    }
```

Comment:
So you just return a new TextComponent with the same string, But we wanted to use the default en_us when translations is off

Reply:
I thought that the default input string was en_us, so I set it as unchanged. 
Is what you want like `TextComponent::Translated(TextComponent::text(message)));`

Reply:
Also, there is no en_us translation file, which is obvious now that i think about it.